dbms/src/Storages/KVStore/Utils/AsyncTasks.h

```diff
@@ -241,6 +258,7 @@ struct AsyncTasks
     // 1. There is already a task registered with the same name and not canceled or fetched.
     bool addTaskWithCancel(Key k, Func f, CancelFunc cf)
     {
+        RUNTIME_CHECK(!shut.load());
```

Comment:
Just return a `false` rather than throwing an exception here is better

Reply:
fixed